stable/kong/templates/deployment.yaml

```diff
@@ -28,14 +28,28 @@ spec:
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         env:
-        - name: KONG_ADMIN_LISTEN
-          value: 0.0.0.0:{{ default 8001 .Values.admin.http.containerPort }}
+        {{- if .Values.admin.useTLS }}
         - name: KONG_ADMIN_LISTEN_SSL
-          value: 0.0.0.0:{{ default 8444 .Values.admin.https.containerPort }}
-        - name: KONG_PROXY_LISTEN
-          value: 0.0.0.0:{{ default 8000 .Values.proxy.http.containerPort }}
+          value: 0.0.0.0:{{ .Values.admin.containerPort }}
+        - name: KONG_ADMIN_SSL
+          value: "on"
+        {{- else }}
+        - name: KONG_ADMIN_LISTEN
+          value: 0.0.0.0:{{ .Values.admin.containerPort }}
+        - name: KONG_ADMIN_SSL
+          value: "off"
+        {{- end }}
+        {{- if .Values.proxy.useTLS }}
         - name: KONG_PROXY_LISTEN_SSL
-          value: 0.0.0.0:{{ default 8443 .Values.proxy.https.containerPort }}
+          value: 0.0.0.0:{{ .Values.proxy.containerPort }}
+        - name: KONG_SSL
+          value: "on"
+        {{- else }}
+        - name: KONG_PROXY_LISTEN
+          value: 0.0.0.0:{{ .Values.proxy.containerPort }}
+        - name: KONG_SSL
+          value: "off"
```

Comment:
What's the purpose of those env vars `KONG_ADMIN_SSL` and `KONG_SSL`?

For the rest, I think it's a good PR, if this is accepted it I'll close #4082 :)

Reply:
[KONG_SSL](https://getkong.org/docs/0.12.x/configuration/#ssl) and [KONG_ADMIN_SSL](https://getkong.org/docs/0.12.x/configuration/#admin_ssl) are Kong configurations.

Reply:
I am waiting for @unguiculus to approve the PR.